src/perf_context.rs

```diff
@@ -53,6 +59,179 @@ pub fn set_perf_level(level: PerfLevel) {
     }
 }
 
+pub struct PerfFlags {
+    inner: NonNull<DBPerfFlags>,
+}
+
+unsafe impl Send for PerfFlags {}
+
+unsafe impl Sync for PerfFlags {}
+
+impl PerfFlags {
+    pub fn new() -> PerfFlags {
+        unsafe {
+            let inner = crocksdb_ffi::crocksdb_create_perf_flags();
+            PerfFlags {
+                inner: NonNull::new_unchecked(inner),
+            }
+        }
+    }
+}
+
+impl Drop for PerfFlags {
+    fn drop(&mut self) {
+        unsafe {
+            crocksdb_ffi::crocksdb_destroy_perf_flags(self.inner.as_ptr());
+        }
+    }
+}
+
+impl Default for PerfFlags {
+    fn default() -> PerfFlags {
+        PerfFlags::new()
+    }
+}
+
+// Generated from librocksdb_sys/src/generate.py and copied here to implement
```

Comment:
do not copy it, how about using a wrapper

Reply:
I think we can't nicely write `PerfFlag::A | PerfFlag::B` that way.

Reply:
Using `,` instead of a `|` doesn't make too much difference. But when using bind-gen, we can hardly be noticing the enum changes and forget to update here.

Reply:
Well, we manually maintain a lot of enums already, I don't know why we made the exception for statistics-related ones in the first place. And I think manual maintaining in general is not a big issue: new version is supposed to be backward compatible, and we would know to update the code if some new variants are truly needed

Reply:
> But when using bind-gen, we can hardly be noticing the enum changes and forget to update here.

I didn't know we actually don't use bind-gen to generate the bindings until I made this change recently. XD

The bindings in `librocksdb_sys/bindings` are just not referenced in `lib.rs`...

Reply:
Yes, the binding is not actually used (the work is paused), another rewrite using bindings is taken in a brand new repo https://github.com/BusyJay/tirocks